### PR TITLE
changes to retrieve disruption stops and routes for services in an organisation

### PR DESCRIPTION
**Pull request by alvak1**
Testing instruction:

Run "npm run dev" from root and then run the below command to invoke and test the api. Update values according to your local env setup

```curl -X GET -H "x-api-key: <your-api-key>" https://<api-gateway-endpoint>/<sst-stage-name>/organisations/<org-id>/impacted/stops```

example:

```curl -X GET -H "x-api-key: <api-key>" https://taejenagw7.execute-api.eu-west-2.amazonaws.com/kalva1/organisations/2c759a03-b961-43e2-9a47-db9a894dfe91/impacted/stops```

### CODE REVIEW

File: packages/organisations-api/get-organisation-stops/index.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+export const main = async (event: APIGatewayEvent): Promise<{ statusCode: number; body: string }> => {
+    try {
+        logger.options.dev = process.env.NODE_ENV !== "production";
+        logger.options.debug = process.env.ENABLE_DEBUG_LOGS === "true" || process.env.NODE_ENV !== "production";
+
+        logger.options.meta = {
+            id: randomUUID(),
+        };
+        logger.info("Starting get organisation stops data...");
```
Comment (alainajaffer) [inline packages/organisations-api/get-organisation-stops/index.ts:183]:
```suggestion
        logger.info("Starting get organisation stops and routes for services data...");
```

File: packages/organisations-api/get-organisation-stops/index.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+        const orgId = event?.pathParameters?.id;
+
+        if (!orgId) {
+            throw new Error("An organisation ID must be provided");
+        }
+
+        const stopsAndServicesMapData = await getOrganisationStops(orgId);
+
+        logger.info("Successfully retrieved organisations DynamoDB...");
```
Comment (alainajaffer) [inline packages/organisations-api/get-organisation-stops/index.ts:199]:
```suggestion
        logger.debug(`Successfully retrieved organisation: ${orgId}'s stops from DynamoDB...`);
```

File: packages/organisations-api/get-organisation-stops/index.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+        return {
+            statusCode: 200,
+            body: JSON.stringify(stopsAndServicesMapData),
+        };
+    } catch (e) {
+        if (e instanceof Error) {
+            logger.error(e);
```
Comment (alainajaffer) [inline packages/organisations-api/get-organisation-stops/index.ts:207]: could we have a message here too like 'Failed to get organisation stops and routes for services data'

File: packages/organisations-api/get-organisation-stops/index.ts [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+            throw new Error("An organisation ID must be provided");
+        }
+
+        const stopsAndServicesMapData = await getOrganisationStops(orgId);
+
+        logger.info("Successfully retrieved organisations DynamoDB...");
+
+        return {
+            statusCode: 200,
+            body: JSON.stringify(stopsAndServicesMapData),
```
[the remark this answers is not in the document]
Comment (alvak1) [reply packages/organisations-api/get-organisation-stops/index.ts:203]: stopsAndServicesMapData will have fields stops and services which will be empty arrays

File: packages/organisations-api/get-organisation-stops/index.ts (hunk 1 above)
Comment (alvak1) [reply packages/organisations-api/get-organisation-stops/index.ts:183]: done

File: packages/organisations-api/get-organisation-stops/index.ts (hunk 2 above)
Comment (alvak1) [reply packages/organisations-api/get-organisation-stops/index.ts:199]: done

File: packages/organisations-api/get-organisation-stops/index.ts (hunk 3 above)
Comment (alvak1) [reply packages/organisations-api/get-organisation-stops/index.ts:207]: done

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**